### DISCUSSION TITLE
Ente Auth @ Android register as URI handler

### DISCUSSION THREAD

**Discussion by grosshad** [ente-io/ente#1845] [Enhancements]
Register as URI handler Inside.Andriod to support URI Sheme

otpauth://TYPE/LABEL?PARAMETERS

See https://github.com/google/google-authenticator/wiki/Key-Uri-Format for more detsils

So you can use any QR reader app to add TOTP Accouts to Ente Auth.

This is seen by many compedive Apps, e.g Google Authenticator, Microsoft Authenticator and Authy just to name some prominent ones

**Comment by ua741** [comment]
We had already added support for this in the past: https://github.com/ente-io/ente/pull/113. Please let us know if it’s not working as intended. We would appreciate it if you could also share the steps to reproduce the issue.

**Comment by grosshad** [comment]
Ok seem that the URI sheme is registered at system level

Unfortunately Motoralas own QR Scanner System App inclued Andriod 14 / Edge 30 neo (using Google Play Services https://play.google.com/store/apps/details?id=com.google.android.gms Version 24.17.18 (190400-633711484) does it own thing to determine witch TOTP App it uses. 

It has the strange behavior that it prefers Microsoft Authentscator (when Authy, Google Authenticator, Microsoft Authenticator and Ente Auth are installed). It neither let me choose or even open the system dialog to temporary use a different App / set System default.

1. removed Microsoft Authenticator
2. next preferred App is Google Authenticator
3. removed Google Authenticator
4. next preferred App is Authy
5. removed Authy
6. Tada now Ente Auth is offered 🤪 

1. restored Microsoft Authenticator
2. now Microsoft Authenticator is preferred again

-> have to use Ente Auth own QR Scanner (that lacks using existing images / Screen shots) / Base32 Setup code aka Shared secret to add more TOTP Accounts

I cannot remove Microsoft Authenticator as it is needed for other reasons too.
